src/teleop/gui/src/components/SATask.vue

```diff
@@ -1,14 +1,330 @@
+      <BasicMap :odom="odom" />
+    </div>
+    <div class="box waypoints light-bg">
+      <BasicWaypointEditor />
+    </div>
+    <div class="box light-bg cameras">
+      <Cameras :primary="true" />
+    </div>
+    <div>
+      <DriveControls />
+    </div>
+    <div class="box light-bg scoop">
+      <EndEffectorUV />
+    </div>
+    <div class="box light-bg arm">
+      <ArmControls />
+    </div>
+    <div class="box light-bg pdb">
+      <PDBFuse />
+    </div>
+    <div class="box light-bg jointState">
+      <JointStateTable :joint-state-data="jointState" :vertical="true" />
+    </div>
+    <div class="box light-bg moteus">
+      <MoteusStateTable :moteus-state-data="moteusState" />
+    </div>
   </div>
 </template>
 
 <script>
+import ROSLIB from "roslib";
+import BasicMap from "./BasicRoverMap.vue";
+import BasicWaypointEditor from "./BasicWaypointEditor.vue";
+import DriveControls from "./DriveControls.vue";
+import EndEffectorUV from "./EndEffectorUV.vue";
+import ArmControls from "./ArmControls.vue";
+import PDBFuse from "./PDBFuse.vue";
+import * as qte from "quaternion-to-euler";
+import Cameras from "./Cameras.vue";
+import MoteusStateTable from "./MoteusStateTable.vue";
+import JointStateTable from "./JointStateTable.vue";
+
 export default {
-  components: {},
+  components: {
+    BasicMap,
+    BasicWaypointEditor,
+    DriveControls,
+    EndEffectorUV,
+    ArmControls,
+    PDBFuse,
+    Cameras,
+    JointStateTable,
+    MoteusStateTable,
+  },
   data() {
-    return {};
+    return {
+      // Default coordinates are at NC 53 Parking Lot
+      odom: {
+        latitude_deg: 42.294864932393835,
+        longitude_deg: -83.70781314674628,
+        bearing_deg: 0,
+      },
+
+      jointState: {},
+      // Moteus state table is set up to look for specific keys in moteusState so it can't be empty
+      moteusState: {
+        name: ["", "", "", "", "", ""],
+        error: ["", "", "", "", "", ""],
+        state: ["", "", "", "", "", ""],
+      },
+
+      // Pubs and Subs
+      odom_sub: null,
+      tfClient: null,
+    };
+  },
+
+  created: function () {
+    this.odom_sub = new ROSLIB.Topic({
+      ros: this.$ros,
+      name: "/gps/fix",
+      messageType: "sensor_msgs/NavSatFix",
+    });
+
+    this.tfClient = new ROSLIB.TFClient({
+      ros: this.$ros,
+      fixedFrame: "odom",
+      // Thresholds to trigger subscription callback
+      angularThres: 0.01,
+      transThres: 0.01,
+    });
+
+    // Subscriber for odom to base_link transform
+    this.tfClient.subscribe("base_link", (tf) => {
+      // Callback for IMU quaternion that describes bearing
+      let quaternion = tf.rotation;
+      quaternion = [quaternion.w, quaternion.x, quaternion.y, quaternion.z];
+      //Quaternion to euler angles
+      let euler = qte(quaternion);
+      // euler[2] == euler z component
+      this.odom.bearing_deg = euler[2] * (180 / Math.PI);
+    });
+
+    this.odom_sub.subscribe((msg) => {
```

Comment:
nit Can we just reorder these so odom_sub subscribes right below where it's defined